src/pfs/pfs.go

```diff
@@ -113,6 +114,9 @@ func (c *Commit) Key() string {
 	if c.Repo == nil {
 		return "<nil repo>@" + c.Id
 	}
+	if (c.Id == "" || strings.ContainsAny(c.Id, ".^")) && c.Branch != nil && c.Branch.Name != "" {
```

Comment:
When using ancestry syntax `. ^ ~`, some code formats the commit as:
id: "^5"
branch: "foo"

So we have to utilize both fields.

Reply:
Since this changes the database key for commits, does there need to be a migration? Relatedly, “some code formats the commit” one way implies other code may format it another way, which implies different code paths may end up using different keys … that seems bad.

Does this have to be `ContainsAny` rather than `HasPrefix`?

`c.Branch != nil && c.Branch.Name != ""` can reduce to `c.Branch.GetName() != ""`: if `c.Branch` is `nil` the `c.Branch.GetName()` will return `""`.

Minor nit: seeing several `if … { return … }` sequences makes me wonder if this would be more legible as a `switch`.